src/components/provider.js

```diff
@@ -23,6 +23,7 @@ const intlFormatPropNames = Object.keys(intlFormatPropTypes);
 const defaultProps = {
     formats : {},
     messages: {},
+    textComponent: 'span',
```

Comment:
This seems good. And if later, in another PR, if there's a good idiomatic way to detect React Native vs React DOM, then this can be updated to be dynamic.